pkg/cluster/arooperator_test.go

```diff
@@ -141,12 +141,60 @@ func TestInstallAROOperator(t *testing.T) {
 		wantErr string
 	}{
 		{
-			name: "install success",
+			name: "install success with service principal",
 			doc: &api.OpenShiftClusterDocument{
 				Key: strings.ToLower(key),
 				OpenShiftCluster: &api.OpenShiftCluster{
 					ID: key,
 					Properties: api.OpenShiftClusterProperties{
+						ServicePrincipalProfile: &api.ServicePrincipalProfile{},
+						IngressProfiles: []api.IngressProfile{
+							{
+								Visibility: api.VisibilityPublic,
+								Name:       "default",
+							},
+						},
+						ProvisioningState: api.ProvisioningStateAdminUpdating,
+						ClusterProfile: api.ClusterProfile{
+							Version: "4.8.18",
+						},
+						NetworkProfile: api.NetworkProfile{
+							APIServerPrivateEndpointIP: "1.2.3.4",
+						},
+					},
+				},
+			},
+			mocks: func(dep *mock_deploy.MockOperator) {
+				dep.EXPECT().
+					Install(gomock.Any()).
+					Return(nil)
+			},
+		},
+		{
+			name: "install success with identity",
+			doc: &api.OpenShiftClusterDocument{
+				Key: strings.ToLower(key),
+				OpenShiftCluster: &api.OpenShiftCluster{
+					ID: key,
+					Identity: &api.Identity{
+						UserAssignedIdentities: api.UserAssignedIdentities{
+							"fakeIdentity": api.ClusterUserAssignedIdentity{
+								ClientID:    "fake",
+								PrincipalID: "alsoFake",
+							},
+						},
+					},
+					Properties: api.OpenShiftClusterProperties{
+						PlatformWorkloadIdentityProfile: &api.PlatformWorkloadIdentityProfile{
+							PlatformWorkloadIdentities: []api.PlatformWorkloadIdentity{
+								{
+									OperatorName: "openshift-azure-operator",
```

Comment:
This test case should fail since the operator name isn't correct (should be `"ServiceOperator"`)... but there's also not much value in me pointing this out given that the unit tests in this file mock the functionality of the operator deployer, which is what this PR actually changes. I think the changes to this file don't add value and can be removed.

With that said, I see why you didn't unit test in `pkg/operator/deploy` though - that `resources` function is tricky to write unit tests for.

To make it easier to unit test the functionality you added in, you could break everything within the `if o.oc.UsesWorkloadIdentity()` out into a helper function that returns `(&corev1.Secret, error)` and then do unit tests for the helper function.

Reply:
Dang, I misunderstood how the install tests worked (clearly!)

I pulled the secret generation code into a helper function and wrote up a unit test for that like you suggested